data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/ResponseExhaustJobV2.scala

```diff
@@ -0,0 +1,110 @@
+      .select(
+        col("userid"),
+        col("courseid"),
+        col("collectionName"),
+        col("batchName"),
+        col("batchid"))
+
+    val batchid = userEnrolmentDataDF.select("batchid").distinct().collect().head.getString(0)
+
+    val assessBlobData = getAssessmentBlobDF(batchid, config)
+
+    val assessAggregateData = loadData(assessmentAggDBSettings, cassandraFormat, new StructType())
+
+    val joinedDF = assessAggregateData.join(assessBlobData, Seq("batch_id", "course_id", "user_id"), "left_outer")
+        .select(assessAggregateData.col("*"))
+
+    joinedDF.join(userEnrolmentDataDF, joinedDF.col("user_id") === userEnrolmentDataDF.col("userid") && joinedDF.col("course_id") === userEnrolmentDataDF.col("courseid"), "inner")
+      .select(userEnrolmentDataDF.col("*"), joinedDF.col("question"), col("content_id"), col("attempt_id"), col("last_attempted_on"))
+      .withColumn("questiondata",explode_outer(col("question")))
+      .withColumn("questionid" , col("questiondata.id"))
+      .withColumn("questiontype", col("questiondata.type"))
+      .withColumn("questiontitle", col("questiondata.title"))
+      .withColumn("questiondescription", col("questiondata.description"))
+      .withColumn("questionduration", round(col("questiondata.duration")))
+      .withColumn("questionscore", col("questiondata.score"))
+      .withColumn("questionmaxscore", col("questiondata.max_score"))
+      .withColumn("questionresponse", UDFUtils.toJSON(col("questiondata.resvalues")))
+      .withColumn("questionoption", UDFUtils.toJSON(col("questiondata.params")))
+      .drop("question", "questiondata", "question_data")
+  }
+
+  def getAssessmentBlobDF(batchid: String, config: JobConfig)(implicit spark: SparkSession): DataFrame = {
+    val azureFetcherConfig = config.modelParams.get("assessmentFetcherConfig").asInstanceOf[Map[String, AnyRef]]
+
+    val store = azureFetcherConfig("store")
+    val format:String = azureFetcherConfig.getOrElse("format", "csv").asInstanceOf[String]
+    val url = store match {
+      case "local" =>
+        val filePath = azureFetcherConfig("filePath").asInstanceOf[String]
+        filePath + s"${batchid}-*.${format}"
+      case "s3" | "azure" =>
+        val bucket = azureFetcherConfig("container").asInstanceOf[String]
+        val key = AppConf.getConfig("azure_storage_key")
+        val filePath = azureFetcherConfig.getOrElse("filePath", "data-archival/").asInstanceOf[String]
+        val file = s"${filePath}${batchid}-*.${format}"
+        s"wasb://$bucket@$key.blob.core.windows.net/$file."
+    }
+    JobLogger.log(s"Fetching data from ${store} for batchid: " + batchid)
+
+    val assessAggData = spark.read.format("csv")
+      .option("header","true")
+      .load(url)
+
+    assessAggData.withColumn("question", extractFromArrayStringFun(col("question")))
+  }
+
+  def extractFromArrayStringFun: UserDefinedFunction =
+    udf { str: String => JSONUtils.deserialize[List[Question]](str) }
```

Comment:
Move this one as well into the UDF util class.

Reply:
Done